crates/matrix-sdk-base/src/client.rs

```diff
@@ -539,7 +539,21 @@ impl BaseClient {
     ) {
         for raw_event in events {
             if let Ok(event) = raw_event.deserialize() {
-                changes.add_room_account_data(room_id, event, raw_event.clone());
+                changes.add_room_account_data(room_id, event.clone(), raw_event.clone());
+
+                // Rooms can either appear in the current request or already be
+                // known to the store. If neither of
+                // those are true then the room is `unknown` and we cannot
+                // process its account data
+                if let AnyRoomAccountDataEvent::MarkedUnread(e) = event {
```

Comment:
`AnyRoomAccountDataEvent::MarkedUnread` exists if and only if the `ruma/unstable-msc2867` feature is enabled in the `crates/matrix-sdk-base/Cargo.toml` file.

Reply:
I've pushed https://github.com/matrix-org/matrix-rust-sdk/pull/3076/commits/bb3f7ede9134a9c49dd23fb822633b9ad14a3989 to solve it.